lib/plugins/aws/package/compile/events/cloud-front.js

```diff
@@ -391,6 +425,16 @@ class AwsCompileCloudFrontEvents {
               shouldAssignCachePolicy = false;
             }
 
+            if (
+              event.cloudFront.behavior &&
+              (event.cloudFront.behavior.ForwardedValues !== undefined ||
+                event.cloudFront.behavior.MaxTTL !== undefined ||
+                event.cloudFront.behavior.MinTTL !== undefined ||
+                event.cloudFront.behavior.DefaultTTL !== undefined)
```

Comment:
Let's construct it more naturally as:

```javascript
event.cloudFront.behavior.ForwardedValues ||
                event.cloudFront.behavior.MaxTTL !=null ||
                event.cloudFront.behavior.MinTTL != null ||
                event.cloudFront.behavior.DefaultTTL !=null
```

Reply:
Prettier transform it to
```js
if (
  (event.cloudFront.behavior &&
  event.cloudFront.behavior.ForwardedValues !== undefined) ||
    event.cloudFront.behavior.MaxTTL !== undefined ||
    event.cloudFront.behavior.MinTTL !== undefined ||
    event.cloudFront.behavior.DefaultTTL !== undefined
)
```

Reply:
Oh ok, I just realized my mistake which is I have to check if `event.cloudFront.behavior` is defined before every `event.cloudFront.behavior.XXX !== undefined` so as not throw `Uncaught TypeError: event.cloudFront.behavior is undefined` Error.

```js
  if (
    event.cloudFront.behavior &&
    event.cloudFront.behavior.ForwardedValues !== undefined ||
    event.cloudFront.behavior &&
    event.cloudFront.behavior.MaxTTL !== undefined ||
    event.cloudFront.behavior &&
    event.cloudFront.behavior.MinTTL !== undefined ||
    event.cloudFront.behavior &&
    event.cloudFront.behavior.DefaultTTL !== undefined
  ) {}
```

Are you sure you prefer this instead of my first solution ?

```js
if (
    event.cloudFront.behavior &&
    (event.cloudFront.behavior.ForwardedValues !== undefined ||
      event.cloudFront.behavior.MaxTTL !== undefined ||
      event.cloudFront.behavior.MinTTL !== undefined ||
      event.cloudFront.behavior.DefaultTTL !== undefined)
  ) {
    console.error("should not throw after apply prettier");
  }
```

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAKVBKABAXgHxbAA6UWWkUAzjFnAG4I06ElllgA2EArgCYBiAJ2gwkLUmzIAjOAAsAhnQCWEQWO5RecAGZKocXqzYBfI8YDcRowHprWfvKUcjS7VlRGy9RgDpOPAWFYHxkFZVUsADJIz1oGYP8+IREQuUUVQR9+VQB3eUEtXgA1eQ5uOEosAEIcZg0tXX1eLAAfFtjvBK4koJhUsIyfAFl5AA8AFXGAGWrarHqdPQNW9okveL7EwJTQ9NVhvUmZmrrNRaaVjo2-bu3g3fDMgBEdeW4OGCPZ04alwwlsMBTFAbHYAArySiUFxuDxrOK+LbJe5pR5RGLwzqbW7IvoPQbZQR5AoGEplCrfeZnRrLNpXRE43r9PaCdH0roBXHMx7DMZfE5U34XOmY65Ipn4iLRdnYzkS1GDIaHaaUhY05oiyRYm5ynYKqUYrVixl6gb7F7aN4fflzNV-IyA4GgrAQqEw9wynU9U0stnw1Da8U+nmE4mFMnlSoCu3C1aSdYM3Uos2ZEYTFXR6l-S7whMc73JlkHKA2n7nWlx+OBk2FnkWq2fDO2rNNDAOwjA4wYTAgAA0IAgAAcYCoqMhQPlhDkIYIEJRkCBSnkAJ7z-tSQTyMAAazgMAAyvIALZwKZLZCWjiUODrzc7vf7wdbvQAc2QMEE5X7cCPMl4hSmeQoBfbh5BfOBCSPeQYBHYCFzeGAID7EBZBgI8OAAdVkJR4EoJ8wDgfc5xwpRlBgZcFzAKFkL0a9BBgMFNxfKCL1Ka9+wAK0oUZ91fDg4AARW4CB4FYq8bxAJ9BDohcpHkGQOGQwdBD0GAMKUXgYFkZAAA4AAZ+2UiBrwwzdBwXZSKjgQQGGQgBHYT4EYod5xQSEAFp9AMAxkNnBylFnRiwJYpBL3YkBryPJR30-CTKD4wTHLgMTwpgeT1M07SkAAJn7D9HA4V8AGEICPEKQAqABWZDuGvcZ5NcsKJLocoAEkzlgfcwBU4cAEFNH3cj+JSuBjGMIA

Reply:
@godu 

```javascript
if ( 
              event.cloudFront.behavior &&
              (event.cloudFront.behavior.ForwardedValues ||
                event.cloudFront.behavior.MaxTTL !=null ||
                event.cloudFront.behavior.MinTTL != null ||
                event.cloudFront.behavior.DefaultTTL !=null)) {
```

Reply:
@medikoo I'm sorry, I misread... It's done !